pkg/controller/hostpathprovisioner/controller.go

```diff
@@ -88,9 +88,20 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 
 	// mapFn will be used to map reconcile requests to the HPP for resources that don't have an ownerRef
 	mapFn := handler.ToRequestsFunc(func(o handler.MapObject) []reconcile.Request {
-		if val, ok := o.Meta.GetLabels()["k8s-app"]; ok && val == "hostpath-provisioner" {
+		if val, ok := o.Meta.GetLabels()["k8s-app"]; ok && val == MultiPurposeHostPathProvisionerName {
+			hppList := &hostpathprovisionerv1.HostPathProvisionerList{}
+
+			if err := mgr.GetClient().List(context.TODO(), hppList, &client.ListOptions{}); err != nil {
+				log.Error(err, "Error listing HPPs")
+				return nil
+			}
+			if len(hppList.Items) == 0 {
+				log.Info("No HPPs in cluster")
+				return nil
+			}
+
```

Comment:
You will want to check for > 1 hpps as well. Which is also an error state.

Reply:
Added this check as well, thx